cassovary-core/src/main/scala/com/twitter/cassovary/util/collections/FastMap.scala

```diff
@@ -0,0 +1,428 @@
+package com.twitter.cassovary.util.collections
+
+import java.util
+
+import it.unimi.dsi.fastutil.ints._
+import it.unimi.dsi.fastutil.objects._
+
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.{specialized => sp}
+
+/**
+ * Specialized, fast and memory efficient implementation of mutable map.
+ *
+ * Provides basic operations only. See [[FastMapWrapper]] for how to use
+ * it as a [[mutable.Map]].
+ *
+ * Use constructor provided in the companion object.
+ */
+trait FastMap[@sp(Int, Long, AnyRef) A, @sp(Int, Long, AnyRef) B] {
+
+  /**
+   * Adds key-value pair to the map.
+   */
+  def +=(key: A, value: B): this.type
+
+  /**
+   * Removes key-value pair with a given key.
+   */
+  def -=(key: A): this.type
+
+  /**
+   * Get is optimized for retrieving elements contained in the Map.
+   *
+   * If the element is not in the map, returns `0` of given type (or false, null).
+   */
+  def get(key: A): B
```

Comment:
My intention was to keep API minimal and give a user guarantee that all the methods are not boxed. If a user wants get with Option, he should use asScala and then he has all the scala's map methods. This separation seems clear to me and combining boxed and pure methods in FastMap may seem confusing...
Beside that to return option we still need to check contains before... that means this method will not be fast and I wanted to avoid that.